lib/pry/history.rb

```diff
@@ -34,6 +34,8 @@ def restore_default_behavior
     # @return [Integer] The number of lines loaded
     def load
       @loader.call do |line|
+        next if invalid_readline_line?(line)
```

Comment:
I wonder if we should substitute `\0` with a blank instead? Like you said in https://github.com/pry/pry/issues/1789 you typed that by accident, right?

Reply:
When you say "substitute `\0` with a blank" do you mean "we could delete `\0`s" or do you mean "we could replace each `\0` with a ` ` (a space)"? I think you might mean the latter, since in #1789 I said I typed `ctrl-space` instead of `space` to generate `\0`. 

I see some issues:
1. In general a null byte bears no relation to a space - `ctrl-space` is _one_ keypress sequence to type a literal NULL byte, for me on Linux, the sequence `ctrl-v, ctrl-shift-'` also enters a NULL byte, and there's likely other ways too
1. There's no guarantee that the NULL byte appears in a position where adding a space is syntactically valid (e.g. `x\0y = 1`)
1. The simple symmetry between "don't save invalid lines" and "don't load existing invalid lines" would be broken

What do you think? I think the behaviour as-is in this PR is simple and likely to be most-often correct vs attempting to modify invalid lines

Reply:
> I think you might mean the latter [...]

That's correct. This is what I meant.

Thanks for the analysis, I agree with this decision. I was just wondering about the "what-if" but your points sound very convincing.